src/Compilers/CSharp/Test/Symbol/Symbols/PartialPropertiesTests.cs

```diff
@@ -3017,11 +3017,879 @@ partial class C
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+
+            var property = comp.GetMember<SourcePropertySymbol>("C.P");
+            AssertEx.Equal([declAttribute, implAttribute], property.GetAttributes().SelectAsArray(a => a.ToString()));
+            AssertEx.Equal([declAttribute, implAttribute], property.PartialImplementationPart!.GetAttributes().SelectAsArray(a => a.ToString()));
+        }
+
+        [Theory]
+        [InlineData("A(1)", "B(2)")]
+        [InlineData("B(2)", "A(1)")]
+        public void Attributes_Property_02(string declAttribute, string implAttribute)
+        {
+            // Lexical order of the partial declarations themselves doesn't affect order that attributes are emitted.
+            var source = $$"""            
+                #pragma warning disable CS9113 // Primary constructor parameter is unread
+
+                using System;
+                
+                class A(int i) : Attribute { }
+                class B(int i) : Attribute { }
+
+                partial class C
+                {
+                    [{{implAttribute}}]
+                    public partial int P { get => 1; set { } }
+
+                    [{{declAttribute}}]
+                    public partial int P { get; set; }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+
+            var property = comp.GetMember<SourcePropertySymbol>("C.P");
+            AssertEx.Equal([declAttribute, implAttribute], property.GetAttributes().SelectAsArray(a => a.ToString()));
+            AssertEx.Equal([declAttribute, implAttribute], property.PartialImplementationPart!.GetAttributes().SelectAsArray(a => a.ToString()));
+        }
+
+        [Fact]
+        public void Attributes_Property_03()
+        {
+            // Order of attributes within a part is preserved.
+            var source = """
+                #pragma warning disable CS9113 // Primary constructor parameter is unread
+
+                using System;
+                
+                [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+                class A(int i) : Attribute { }
+                [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
+                class B(int i) : Attribute { }
+
+                partial class C
+                {
+                    [A(2), B(2)]
+                    public partial int P { get => 1; set { } }
+
+                    [A(1), B(1)]
+                    public partial int P { get; set; }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+
+            var property = comp.GetMember<SourcePropertySymbol>("C.P");
+            AssertEx.Equal(["A(1)", "B(1)", "A(2)", "B(2)"], property.GetAttributes().SelectAsArray(a => a.ToString()));
```

Comment:
>.SelectAsArray(a => a.ToString()) [](http://example.com/codeflow?start=85&length=33)

nit: consider adding a `ToStrings()` extension for `ImmutableArray<CSharpAttributeData>`. It could go into `Extensions.cs` along with other extensions related to `CSharpAttributeData` (such as `VerifyNamedArgumentValue`).